### ISSUE TITLE
Projection formulas are (almost definitely) incorrect

### ISSUE THREAD

**Issue by achalddave** [jysung/activity_detection#1]
I've been trying to use these projection formulas: https://github.com/jysung100/activity_detection/blob/master/FeatureExtractor/featuresRGBD.cpp#L26

``` cpp
  // Given (x,y,z) coordinates, converts that point into its x pixel number in the 2D image.
  int xPixelFromCoords(double x, double y, double z)
  {
    return (int) (156.8584456124928 + 0.0976862095248 * x - 0.0006444357104 * y + 0.0015715946682 * z);
  }

  // Given (x,y,z) coordinates, converts that point into its y pixel number in the 2D image.
  int yPixelFromCoords(double x, double y, double z)
  {
    return (int) (125.5357201011431 + 0.0002153447766 * x - 0.1184874093530 * y - 0.0022134485957 * z);
  }
```

These purport to go from the joint's 3d coordinates to 2d coordinates, but they do not seem to work. I visualized the 2d joint locations after this projection, and they are significantly off; furthermore, it seems odd that you could project from 3d to 2d with a linear dot product (since projections should be scaled by z). 

The correct formula should be as per OpenNI's formula, which is discussed by a developer here: https://groups.google.com/d/msg/unitykinect/1ZFCHO9PpjA/1KdxUTdq90gJ (Note that there is a minor mistake in the algebra, as I showed here: https://groups.google.com/d/msg/unitykinect/1ZFCHO9PpjA/0B2eHjrywj4J).

I'm however not sure if this function serves a different purpose, so I am not submitting a pull request. Further, this translation from 3d to 2d is not documented in CAD120 anywhere; it would be nice to incorporate the formula from that post into the README for the skeleton format to make 2d visualizations, among other things, easier.

**Comment by jysung** [comment]
You are correct that this formula is incorrect. I just didn't had a chance to fix it yet.
If you can create a pull request, it will be great!

Thanks,
Jae

**Comment by achalddave** [comment]
Thanks for the quick reply! Will send one later tonight or tomorrow.

Edit: I've committed the changes to my fork, but haven't had a chance to test it yet (not on a dev machine for a bit); will submit PR once I am.

**Comment by raminegdr** [comment]
could you  release demo of feature extraction?or one of sample your dataset?
